pydeseq2/dds.py

```diff
@@ -288,12 +272,12 @@ def fit_genewise_dispersions(self):
                         batch_size=self.batch_size,
                     )(
                         delayed(fit_lin_mu)(
-                            counts=counts_nonzero[:, i],
-                            size_factors=size_factors,
+                            counts=self.X[:, i],
+                            size_factors=self.obsm["size_factors"],
                             design_matrix=X,
                             min_mu=self.min_mu,
```

Comment:
Cf above comment: these lines are confusing as `self.X` and `X` actually contain very different kinds of quantities